app/models/replacement_document_validation_request.rb

```diff
@@ -30,6 +30,19 @@ def reset_document_invalidation
     raise ResetDocumentInvalidationError, e.message
   end
 
+  def replace_document!(file:, reason:)
+    transaction do
+      self.new_document = planning_application.documents.create!(
+        file: file,
+        tags: old_document.tags,
+        numbers: old_document.numbers
+      )
+
+      close!
```

Comment:
What are we closing? Is it the new_document so we can change the old_document .... ?? confused

Reply:
The request!

Reply:
Just as well I asked. So the request is for the new document ... you've replace the old document, you close the request and then you update the new document with info. 👍